### PR TITLE
Improve readability of error messages in Camel K

### PR DESCRIPTION
**Pull request by orpiske**
Colorize pod log messages based on their log level to improve
readability of errors, warnings and pod output messages

**Release Note**
```release-note
Colorize pod log messages based on their log level
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by orpiske** [comment]
Samples for the logs with colorization: 

With errors on the pods/build:
colorize-2-with-errors

With a pod output (ie.: when the pod uses `System.out.println`):
colorize-2-with-sout

With a basic pod output (ie.: using `to("log:info")`):
colorize-2-working

**Comment by orpiske** [comment]
The idea is to propose a way to improve spotting build errors (as suggested on #185)

**Comment by astefanutti** [comment]
Nice. I wonder, does it support https://no-color.org?

**Comment by orpiske** [comment]
> Nice. I wonder, does it support https://no-color.org?

Thanks. No, but I can include that.

**Comment by orpiske** [comment]
> Nice. I wonder, does it support https://no-color.org?

Done. I also added a small note on the readme for that.

**Comment by astefanutti** [comment]
Out of curiosity, I wonder if we could enable colors in the integration logs directly. IIRC, it used to be enabled with Spring Boot, when we supported it as runtime. Is this possible with Quarkus? If yes, that would benefit globally, and avoid client parsing altogether.

**Comment by orpiske** [comment]
> Out of curiosity, I wonder if we could enable colors in the integration logs directly. IIRC, it used to be enabled with Spring Boot, when we supported it as runtime. Is this possible with Quarkus? If yes, that would benefit globally, and avoid client parsing altogether.

IIRC, yes. I can take a look at this route, as well. I am not sure how it would work w/ displaying the logs via kubectl though. In this case it is done on the client side, so if the user used kubectl / oc to read the logs, there is no impact.

**Comment by orpiske** [comment]
> Out of curiosity, I wonder if we could enable colors in the integration logs directly. IIRC, it used to be enabled with Spring Boot, when we supported it as runtime. Is this possible with Quarkus? If yes, that would benefit globally, and avoid client parsing altogether.

This works: `kamel run --property quarkus.log.console.color=true Basic.java`. So, that's an even easier alternative. Maybe we could make this property enabled by default instead of doing it client side? wdyt?

**Comment by orpiske** [comment]
logging-via-quarkus

**Comment by astefanutti** [comment]
Very nice! Yes, it'd +1 to enabled this by default. Maybe we can add a `noColor` option to the Quarkus trait to have quick way to disable it.

Have you tested the behaviour of `kubectl logs`?

**Comment by orpiske** [comment]
> Very nice! Yes, it'd +1 to enabled this by default. Maybe we can add a `noColor` option to the Quarkus trait to have quick way to disable it.

Awesome! I'll take a look at that. It seems much better than the original approach.

> 
> Have you tested the behaviour of `kubectl logs`?

Yes I did. It works just as nice via `kubectl`. Also, there is no visual harm when viewing them via the UI:

Screenshot_2021-04-29 Kubernetes Dashboard

I think it should be reasonably safe. 

One last benefit is that doing it via the quarkus property also enables it on the `kamel local run` (which was not supported originally).

**Comment by lburgazzoli** [comment]
maybe we can set this logging option via an env var when the integration is deployed, so the user has the option to override it using properties (or env var)

**Comment by astefanutti** [comment]
Awesome! Maybe it is a good opportunity to introduce a logging trait: #580?

**Comment by orpiske** [comment]
> Awesome! Maybe it is a good opportunity to introduce a logging trait: #580?

I'm not sure if my golang skills are up to the task, but I can give it a try too :)

**Comment by orpiske** [comment]
@astefanutti @lburgazzoli Here we go. This adds a new trait `logging` to control turning on/off the color. It's very rudimentary for now, but I believe we could extend it/adjust it for #580. 

Currently the logging is adjusted by injecting an environment variable `QUARKUS_LOG_CONSOLE_COLOR` with the appropriate value as set via `--trait logging.color=(true/false)` .

File: pkg/trait/logging.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	envVarQuarkusLogConsoleColor	= "QUARKUS_LOG_CONSOLE_COLOR"
+)
+
+
+// This trait is used to control logging options (such as color)
+//
+// +camel-k:trait=logging
+type loggingTrait struct {
+	BaseTrait `property:",squash"`
+	Color *bool `property:"color" json:"color,omitempty"`
```
Comment (astefanutti) [inline pkg/trait/logging.go:37]: Would be useful to have a one line description for the generated documentation.

**Review by astefanutti — APPROVED** [review]
LGTM! Could you run `go fmt` and re-generate the traits documentation with `make generate-doc`.

File: pkg/trait/logging.go (hunk 1 above)
Comment (orpiske) [reply pkg/trait/logging.go:37]: Sure thing, I'll do that!

**Review by astefanutti — APPROVED** [review]

**Pull request merged**